packages/flutter_tools/lib/src/commands/start.dart

```diff
@@ -204,6 +211,34 @@ Future<int> startApp(
   return startedSomething ? 0 : 2;
 }
 
+/// Delay until the Observatory / service protocol is available.
+///
+/// This does not fail if we're unable to connect, and times out after the given
+/// [timeout].
+Future _delayUntilObservatoryAvailable(String host, int port, {
+  Duration timeout: const Duration(seconds: 10)
+}) async {
+  Stopwatch stopwatch = new Stopwatch()..start();
+
+  final String url = 'ws://${host}:${port}/ws';
+  printTrace('Looking for the observatory at $url.');
+
+  while (stopwatch.elapsed <= timeout) {
+    try {
+      printTrace('Attempting to connect to the observatory port...');
```

Comment:
Yeah, it's tracing. When developing this, it was useful to know how many times through the loop we had to do before we connected. I can remove, since the time is already printed out in verbose mode -